tests/load/gcp/lma.tpl.conf

```diff
@@ -0,0 +1,133 @@
+    kind: Ingress
+    metadata:
+      name: prom-ingress
+      namespace: lma-load-test
+      annotations:
+        nginx.ingress.kubernetes.io/use-regex: "true"
+        nginx.ingress.kubernetes.io/rewrite-target: /$1
+    spec:
+      ingressClassName: public
+      rules:
+      - http:
+          paths:
+          - path: /prom/(.*)
+            pathType: Prefix
+            backend:
+              service:
+                name: prometheus
+                port:
+                  number: 9090
+          - path: /loki/(.*)
+            pathType: Prefix
+            backend:
+              service:
+                name: loki
+                port:
+                  number: 3100
+- path: /run/overlay-load-test.yaml
+  content: |
+    --- # overlay.yaml
+    applications:
+      scrape-target:
+        charm: prometheus-scrape-target-k8s
+        scale: 1
+        trust: true
+        channel: edge
+        options:
+          targets: ${join(",", [for port in PORTS: "${AVALANCHE_URL}:${port}"])}
+      scrape-config:
+        charm: prometheus-scrape-config-k8s
+        scale: 1
+        trust: true
+        channel: edge
+        options:
+          scrape-interval: ${SCRAPE_INTERVAL}s
+          scrape-timeout: ${max(SCRAPE_INTERVAL - 1, 1)}s
+
+    relations:
+    - - scrape-target:metrics-endpoint
+      - scrape-config:configurable-scrape-jobs
+    - - scrape-config:metrics-endpoint
+      - prometheus:metrics-endpoint
+
+package_update: true
+
+packages:
+- python3-pip
+- jq
+
+snap:
+  commands:
+  - snap install --classic juju
+  - snap install --classic microk8s
+  - snap alias microk8s.kubectl kubectl
+
+runcmd:
+- |
+  # disable swap
+  sysctl vm.swappiness=0
+  echo "vm.swappiness = 0" | tee -a /etc/sysctl.conf
+
+- |
+  # disable unnecessary services
+  systemctl disable man-db.timer man-db.service --now
+  systemctl disable apport.service apport-autoreport.service  --now
+  systemctl disable apt-daily.service apt-daily.timer --now
+  systemctl disable apt-daily-upgrade.service apt-daily-upgrade.timer --now
+  systemctl disable unattended-upgrades.service --now
+  systemctl disable motd-news.service motd-news.timer --now
+  systemctl disable bluetooth.target --now
+  systemctl disable ua-messaging.service ua-messaging.timer --now
+  systemctl disable systemd-tmpfiles-clean.timer --now
+
+- |
+  # setup microk8s and bootstrap
+  adduser ubuntu microk8s
+  microk8s status --wait-ready
+  microk8s enable dns:$(grep nameserver /run/systemd/resolve/resolv.conf | awk '{print $2}')
+  microk8s.enable storage ingress
+  # wait for addons to become available
+  sh -c "until microk8s.kubectl rollout status deployments/hostpath-provisioner -n kube-system -w; do sleep 5; done"
+  sh -c "until microk8s.kubectl rollout status deployments/coredns -n kube-system -w; do sleep 5; done"
+  sh -c "until microk8s.kubectl rollout status daemonsets/nginx-ingress-microk8s-controller -n ingress -w; do sleep 5; done"
```

Comment:
I still think you should drop this loop and issue a time out of something like 10 minutes, combined with setting `set -euo pipefail` at the top of this script. I'd rather this test failed explicitly and the instance was cleaned up with Terraform automatically than have the chance of the instance just running forever...

You can use `set -euxo pipefail` if you want more verbose output (`set -x` will show each of the commands in the output as the script is executed)

Reply:
It's not a sentiment against loops - I just don't see the point of this being in what is essentially an infinite while loop? 

By using `kubectl rollout` with a timeout (as suggested in a previous review), if the timeout is exceeded, the return code will be >0, and thus the script will terminate when `set -euo pipefail` is used. I don't recall exactly how cloud init behaves when one piece fails, but worth looking into.

Thoughts @simskij?

Reply:
(For reference, previous suggesting was to use something like: `kubectl rollout status daemonset.apps/speaker -n metallb-system -w --timeout=600s`)

Reply:
Ah okay, you might be able to make `cloud-init` run it using bash by adding a shebang at the start? Something like:

```yaml
runcmd:
- |
    #!/bin/bash
    set -euxo pipefail
    
    #stuff
```

I don't know if that'll work, but useful if it does. No need in scripts that aren't using pipes, obviously.

IIRC cloud-init does this automatically in user-data scripts.

Reply:
I believe that user-data calls the same `shellify` method as normal, which is still `/bin/sh`, but a shebang specifying bash definitely does work.

Reply:
Adding the shebang does not help
`set: Illegal option -o pipefail`
That is because the autogenerated `/var/lib/cloud/instance/scripts/runcmd` already has `#!/bin/sh` at the top, so my `#!/bin/bash` is just a plain comment.

Reply:
Heh. No worries then. Thanks for updating to use the timeout.